internal/search/graphql.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright (c) 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package search
```

Comment:
I think it would be good to keep in the "search" package only the code related to search mechanisms defined in the specification, independent of the implementation of the backend. Can we move this code to a separate package? For example `internal/graphql`?